model/src/test/test_differenced_process.py

```diff
@@ -0,0 +1,122 @@
+    that valid types do pass validation, and
+    that this function is correctly used for
+    type validation at object instantiation.
+    """
+    err_match = "must be an integer"
+    with pytest.raises(ValueError, match=err_match):
+        DifferencedProcess.assert_valid_differencing_order(wrong_type_order)
+    with pytest.raises(ValueError, match=err_match):
+        _ = DifferencedProcess(
+            "should_fail",
+            fundamental_process=None,
+            differencing_order=wrong_type_order,
+        )
+    DifferencedProcess.assert_valid_differencing_order(1)
+    _ = DifferencedProcess(
+        "should_succeed", fundamental_process=None, differencing_order=1
+    )
+
+
+@pytest.mark.parametrize(
+    ["wrong_value", "right_value"], [[0, 1], [-5, 5], [-10325235, 300]]
+)
+def test_differencing_order_value_validation(wrong_value, right_value):
+    """
+    Test that passing an integer that is less than 1
+    as the differencing_order raises a ValueError via
+    the dedicated assertion function, that valid
+    values do pass, and that the validation function
+    is correctly used for value validation at
+    object instantiation.
+    """
+    with pytest.raises(ValueError, match="greater than or equal to 1"):
+        DifferencedProcess.assert_valid_differencing_order(wrong_value)
+        _ = DifferencedProcess(
+            "should_fail",
+            fundamental_process=None,
+            differencing_order=wrong_value,
+        )
+
+    DifferencedProcess.assert_valid_differencing_order(right_value)
+    _ = DifferencedProcess(
+        "should_succeed",
+        fundamental_process=None,
+        differencing_order=right_value,
+    )
+
+
+@pytest.mark.parametrize(
+    ["order", "diffs"],
+    [
+        [1, jnp.array([1.0, 2, -3])],
+        [2, jnp.array([1.0, 2, -3])],
+        [3, jnp.array([1.0, 2, -3])],
+        [4, jnp.array([1.0, 2, -3])],
+    ],
+)
+def test_integrator_init_validation(order, diffs):
+    """
+    Test that when the integrator is called,
+    it succeeds if and only if the right number
+    of initial values have been specified, and raises
+    the appropriate ValueError otherwise.
+    """
+    inits_short = jnp.ones(order - 1)
+    inits_correct = jnp.ones(order)
+    inits_long = jnp.ones(order + 1)
+    proc = DifferencedProcess(
+        f"test_difference_process_of_order_{order}",
+        fundamental_process=None,
+        differencing_order=order,
+    )
+    with pytest.raises(
+        ValueError, match="exactly as many initial difference values"
+    ):
+        proc.integrate(inits_short, diffs)
+    with pytest.raises(
+        ValueError, match="exactly as many initial difference values"
+    ):
+        proc.integrate(inits_long, diffs)
+    proc.integrate(inits_correct, diffs)
+
+
+@pytest.mark.parametrize(
+    ["order", "n_diffs"], [[1, 250], [2, 40], [3, 10], [4, 10], [5, 5]]
+)
+def test_integrator_correctness(order, n_diffs):
```

Comment:
A non-random example computed by hand would be illustrative.

Reply:
addressed in 0bfc0d3